src/WalletLink.ts

```diff
@@ -185,10 +173,21 @@ export class WalletLink {
    * all potential stale state is cleared.
    */
   public disconnect(): void {
-    if (typeof window.walletLinkExtension !== "undefined") {
-      window.walletLinkExtension.close()
+    const extension = this.getCBWalletExtension()
+    if (extension) {
+      extension.close()
     } else {
       this._relay?.resetAndReload()
     }
   }
+
+  private getCBWalletExtension(): WalletLinkProvider | undefined {
```

Comment:
Would a getter be better than a private method?

```ts
get walletExtenstion () {
    return window.walletLinkExtension
}
```

Reply:
nice